pkg/pipeline/pipeline_builder.go

```diff
@@ -260,24 +281,31 @@ func (b *builder) getStageNode(pe *pipelineEntry, stageID string) (interface{},
 		b.terminalNodes = append(b.terminalNodes, encode)
 		stage = encode
 	case StageTransform:
-		stage = node.AsMiddle(func(in <-chan []config.GenericMap, out chan<- []config.GenericMap) {
+		stage = node.AsMiddle(func(in <-chan config.GenericMap, out chan<- config.GenericMap) {
 			b.opMetrics.CreateInQueueSizeGauge(stageID, func() int { return len(in) })
 			b.opMetrics.CreateOutQueueSizeGauge(stageID, func() int { return len(out) })
 			for i := range in {
 				b.runMeasured(stageID, func() {
-					out <- pe.Transformer.Transform(i)
+					if transformed, ok := pe.Transformer.Transform(i); ok {
+						out <- transformed
+					}
 				})
 			}
 		})
 	case StageExtract:
-		stage = node.AsMiddle(func(in <-chan []config.GenericMap, out chan<- []config.GenericMap) {
+		stage = node.AsMiddle(func(in <-chan config.GenericMap, out chan<- config.GenericMap) {
 			b.opMetrics.CreateInQueueSizeGauge(stageID, func() int { return len(in) })
 			b.opMetrics.CreateOutQueueSizeGauge(stageID, func() int { return len(out) })
-			for i := range in {
-				b.runMeasured(stageID, func() {
-					out <- pe.Extractor.Extract(i)
-				})
-			}
+			// TODO: replace batcher by rewriting the different extractor implementations
+			// to keep the status while processing flows one by one
+			utils.Batcher(utils.ExitChannel(), b.batchMaxLen, b.batchTimeout, in,
```

Comment:
Didn't provide unit tests for Batcher as it is already tested as an internal component in the conntrack integration tests.

Reply:
Nice idea to build the Batcher in this way for all the stages that need it!